pkg/controller/routes.go

```diff
@@ -47,7 +47,7 @@ func routes(router *router.Router, registryTransport http.RoundTripper) {
 	router.HandleFunc(&v1.AppInstance{}, images.CreateImages)
 	router.HandleFunc(&v1.AppInstance{}, appdefinition.ParseAppImage)
 	router.HandleFunc(&v1.AppInstance{}, tls.ProvisionCerts) // Provision TLS certificates for port bindings with user-defined (valid) domains
-	router.Type(&v1.AppInstance{}).Middleware(appdefinition.FilterLabelsAndAnnotationsConfig).HandlerFunc(namespace.AddNamespace)
+	router.Type(&v1.AppInstance{}).Middleware(appdefinition.FilterLabelsAndAnnotationsConfig, appdefinition.RefreshStatus).HandlerFunc(namespace.AddNamespace)
```

Comment:
This middleware should be added on line 56 instead. By adding it here, before we calculate the defaults, then the defaults won't get recalculated if the generation doesn't change. The reason this is working right now is that you are actually changing the generation.